src/pkg/caendr/caendr/api/strain.py

```diff
@@ -170,14 +170,9 @@ def get_bam_bai_download_link(species, strain_name, ext, signed=False):
   '''
 
   bucket_name = MODULE_SITE_BUCKET_PRIVATE_NAME
-  bam_prefix = BAM_BAI_PREFIX.get_string(SPECIES=species.name)
+  bam_prefix  = BAM_BAI_PREFIX.get_string(SPECIES=species.name)
 
-  blob_name = f'{bam_prefix}/{strain_name}.{ext}'
-
-  if signed:
-    return generate_download_signed_url_v4(bucket_name, blob_name)
-  else:
-    return generate_blob_url(bucket_name, blob_name)
+  return generate_blob_uri( bucket_name, bam_prefix, f'{strain_name}.{ext}', BlobURISchema.sign(signed) )

```

Comment:
The new unified `generate_blob_uri` function can handle optionally signing a URL, so no need to swap between two different functions here